src/__internal__/focus-trap/focus-trap.test.tsx

```diff
@@ -238,723 +178,727 @@ describe("FocusTrap", () => {
       });
 
       it("should back to the first item when use `shift + tab`", async () => {
-        focusElement(screen.getByText(BUTTON_TWO));
-        await shiftTabPress();
-        expect(screen.getByText(BUTTON_ONE)).toHaveFocus();
+        const { user } = setup(mockComponentToRender());
+        const buttonTwo = screen.getByRole("button", { name: "Two" });
+        buttonTwo.focus();
+
+        await user.tab({ shift: true });
+
+        const buttonOne = screen.getByRole("button", { name: "One" });
+        expect(buttonOne).toHaveFocus();
       });
 
       it("should go to the second item when use TAB", async () => {
-        focusElement(screen.getByText(BUTTON_ONE));
-        await tabPress();
-        expect(screen.getByText(BUTTON_TWO)).toHaveFocus();
+        const { user } = setup(mockComponentToRender());
+        const buttonOne = screen.getByRole("button", { name: "One" });
+        buttonOne.focus();
+
+        await user.tab();
+
+        const buttonTwo = screen.getByRole("button", { name: "Two" });
+        expect(buttonTwo).toHaveFocus();
       });
 
       it("should move to the first focusable item if TAB pressed on last focusable item", async () => {
-        focusElement(screen.getByText(BUTTON_TWO));
-        await tabPress();
-        expect(screen.getByText(BUTTON_ONE)).toHaveFocus();
+        const { user } = setup(mockComponentToRender());
+        const buttonTwo = screen.getByRole("button", { name: "Two" });
+        buttonTwo.focus();
+
+        await user.tab();
+
+        const buttonOne = screen.getByRole("button", { name: "One" });
+        expect(buttonOne).toHaveFocus();
       });
     });
 
     describe("and element does not have focusable items", () => {
-      beforeEach(() => {
-        render(mockComponentToRender({ children: <p>Test content</p> }));
-      });
-
       it("should block tabbing if `tab` pressed", async () => {
-        expect(screen.getByTestId(WRAPPER_ID)).toHaveFocus();
-        await tabPress();
-        expect(screen.getByTestId(WRAPPER_ID)).toHaveFocus();
+        const { user } = setup(<MockComponent>Test content</MockComponent>);
+        expect(screen.getByRole("dialog")).toHaveFocus();
+        await user.tab();
+        expect(screen.getByRole("dialog")).toHaveFocus();
       });
 
       it("should block shift tabbing if `shift + tab` is pressed", async () => {
-        expect(screen.getByTestId(WRAPPER_ID)).toHaveFocus();
-        await shiftTabPress();
-        expect(screen.getByTestId(WRAPPER_ID)).toHaveFocus();
+        const { user } = setup(<MockComponent>Test content</MockComponent>);
+        expect(screen.getByRole("dialog")).toHaveFocus();
+        await user.tab({ shift: true });
+        expect(screen.getByRole("dialog")).toHaveFocus();
       });
     });
 
     describe("and some children elements are disabled", () => {
-      beforeEach(() => {
-        render(
+      it("only focuses those that are not", async () => {
```

Comment:
suggestion(non-blocking): be more verbose with language

that are not what? This is more an issue with the describe block which we obviously need to revisit with these files but still think since we're adding an `it` it would make sense to be more clear